plotting/ft_plot_vector.m

```diff
@@ -337,19 +337,12 @@
         % plot each line with its own color
         h = plot(hdat, vdat(i,:), style, 'LineWidth', linewidth, 'Color', color(i), 'markersize', markersize, 'markerfacecolor', markerfacecolor);
       end
-      if ~isempty(parent)
```

Comment:
if the variable h is recycled for each line to be plotted, I'd say that indeed it doesn't make sense to add the parent property to the handle, if it's to be overwritten in the next iteration. Yet, what I don't understand in general from this part of the code, is why there is a loop across i, but i is only used in line 338 and not in the other situations. Does this mean that with matricial input the matrix is plotted i times on top of itself?

Reply:
Ah, it seems that this is incorrect only here. further down in the code the handles are indeed correctly updated in the loop, i.e. h(i) rather than h. I would assume that this also needs to be done here.

In that case (with a vector (or even matrix) of handles) I'd assume that the parent needs to be added to each of them. This can be done once at the bottom of the function indeed, but may require a for-loop there.

Reply:
OK, overall it seems that the section that deals with the saturation case is messy.
in case multiple handles are needed (i.e. with multiple colors and/or with multiple saturated segments) the handle currently overwrites itself, which is probably not intended. Also the for loop across i is misplaced.

While you're at it, could I ask you to address this inconsistency?